pkg/suggestion/v1beta1/nas/darts/service.py

```diff
@@ -134,3 +140,60 @@ def get_algorithm_settings(settings_raw):
         algorithm_settings_default[s_name] = s_value
 
     return algorithm_settings_default
+
+
+def validate_algorithm_settings(spec: api_pb2.ExperimentSpec) -> (bool, str):
+    # Validate Operations
+    is_valid, message = validate_operations(spec.nas_config.operations.operation)
+    if not is_valid:
+        return False, message
+
+    # Validate AlgorithmSettings
+    is_valid, message = validate_algorithm_spec(spec.algorithm.algorithm_settings)
+    if not is_valid:
+        return False, message
+
+    return True, ""
+
+
+def validate_algorithm_spec(algorithm_settings: list[api_pb2.AlgorithmSetting]) -> (bool, str):
```

Comment:
Thank you for pointing that out. @anencore94 
I implemented the validation based on [`quark0/darts`](https://github.com/quark0/darts) and [`pt.darts`](https://github.com/khanrc/pt.darts), which is the same as the core logic of darts suggestion service, so we might not need mention it additionally in the readme. Although, it might be better to add comments to source codes for developers.

> The implementation is based on [official github implementation](https://github.com/quark0/darts) and [popular repository](https://github.com/khanrc/pt.darts).

https://github.com/kubeflow/katib/tree/master/pkg/suggestion/v1beta1/nas/darts#about-the-differentiable-architecture-search

Reply:
Thanks !!

Reply:
Done.